pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -217,15 +241,21 @@ func (nm *NodeManager) removeNode(node *v1.Node) {
 }
 
 func (nm *NodeManager) GetNodeInfo(nodeName k8stypes.NodeName) (NodeInfo, error) {
-	nm.nodeInfoLock.RLock()
-	nodeInfo := nm.nodeInfoMap[convertToString(nodeName)]
-	nm.nodeInfoLock.RUnlock()
+	getNodeInfo := func (nodeName k8stypes.NodeName) (*NodeInfo) {
+		nm.nodeInfoLock.RLock()
+		nodeInfo := nm.nodeInfoMap[convertToString(nodeName)]
+		nm.nodeInfoLock.RUnlock()
+		return nodeInfo
+	}
+	nodeInfo := getNodeInfo(nodeName)
 	if nodeInfo == nil {
 		err := nm.RediscoverNode(nodeName)
 		if err != nil {
-			return NodeInfo{}, fmt.Errorf("error %q node info for node %q not found", err, convertToString(nodeName))
+			glog.V(4).Infof("error %q node info for node %q not found", err, convertToString(nodeName))
+			return NodeInfo{}, err
 		}
 	}
+	nodeInfo = getNodeInfo(nodeName)
```

Comment:
Consider nodeInfo is not nil, you are making call to getNodeInfo(nodeName) twice. Shouldn't you do it only if nodeInfo is nil.

Reply:
Good Catch! Will fix this